src/platform/psvita/psp2_ui.cpp

```diff
@@ -71,75 +71,116 @@ namespace {
 	vita2d_texture* gpu_texture;
 	vita2d_texture* next_texture;
 	vita2d_texture* main_texture;
+	vita2d_texture* touch_texture;
 	uint8_t zoom_state;
 	int in_use_shader;
 	bool set_shader;
+	bool zoom_trigger;
+	SceUID GPU_Thread;
 	SceUID GPU_Mutex, GPU_Cleanup_Mutex;
+	bool is_pstv = false;
+
+	const int touch_left[] = {
+		Input::Keys::N1, Input::Keys::N2, Input::Keys::N3, Input::Keys::N4,
+		Input::Keys::N5, Input::Keys::N6, Input::Keys::N7, Input::Keys::N8
+	};
+
+	const int touch_right[] = {
+		Input::Keys::ESCAPE, Input::Keys::N9, Input::Keys::N0,
+		Input::Keys::KP_PERIOD, Input::Keys::KP_ADD, Input::Keys::KP_SUBTRACT,
+		Input::Keys::KP_MULTIPLY, Input::Keys::KP_DIVIDE
+	};
+
+	constexpr int SCREEN_WIDTH = 960;
+	constexpr int SCREEN_HEIGHT = 544;
+
+	constexpr int touch_buttons_width = 117;
+	constexpr int touch_buttons_height = SCREEN_HEIGHT/8;
+	constexpr int touch_buttons_right_x = SCREEN_WIDTH - touch_buttons_width;
+	std::vector<bool> touched_buttons(16, false);
 }
 
 static int renderThread(unsigned int args, void* arg){
-	
 	for (;;){
-	
-		sceKernelWaitSema(GPU_Mutex, 1, NULL);
-		memcpy(vita2d_texture_get_datap(gpu_texture), vita2d_texture_get_datap(next_texture), vita2d_texture_get_stride(gpu_texture)*240);
+
+		sceKernelWaitSema(GPU_Mutex, 1, nullptr);
+		memcpy(vita2d_texture_get_datap(gpu_texture),
+			vita2d_texture_get_datap(next_texture),
+			vita2d_texture_get_stride(gpu_texture)*240);
```

Comment:
wrong indent

Reply:
I indented all function arguments spanning over multiple lines with one additional tab, we really need a policy for this (coding style).

Reply:
oh sorry I did not see this is part of the argument list :facepalm: